x/pairing/keeper/grpc_query_user_entry.go

```diff
@@ -31,9 +31,33 @@ func (k Keeper) UserEntry(goCtx context.Context, req *types.QueryUserEntryReques
 	// support legacy
 	project, vrfpk_proj, err := k.GetProjectData(ctx, userAddr, req.ChainID, epochStart)
 	if err == nil {
-		allowedCU := project.Policy.EpochCuLimit
+		plan, err := k.subscriptionKeeper.GetPlanFromSubscription(ctx, project.GetSubscription())
```

Comment:
lets return the user entry as it is, no need to check all those cases

Reply:
I can't do that. If the user is not legacy, his geolocation is determined by the policies it obeys to. If he's legacy, the geolocation is determined by his stake entry. Note that this is only an example, there are more problematic fields. Please explain what do you mean

Reply:
i was not clear, we dont need to check the policies here, everything else is good :)

Reply:
ok. Deleted the VerifyProject() call